engine/src/main/java/org/terasology/engine/rendering/logic/ChunkMeshRenderer.java

```diff
@@ -51,7 +51,7 @@ public void onNewMesh(OnActivatedComponent event, EntityRef entity, ChunkMeshCom
         disposalSet.add(new BuffersReference(mesh.mesh, disposalQueue));
     }
 
-    @ReceiveEvent(components = {MeshComponent.class, LocationComponent.class})
+    @ReceiveEvent(components = {ChunkMeshComponent.class, LocationComponent.class})
```

Comment:
BeforeDeactivateComponent is called when a component is removed from an entity. you would get this hit if either LocationComponent or ChunkMeshComponent is removed. I guess in either case this would be valid behavior? would help to have some unit test verify the behavior. the fix is good as it is at the moment.